pkg/imgutil/dockerconfigresolver/credentialsstore.go

```diff
@@ -0,0 +1,182 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package dockerconfigresolver
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/docker/cli/cli/config/types"
+
+	"github.com/containerd/log"
+)
+
+type Credentials = types.AuthConfig
```

Comment:
Suggest change to `type AuthConfig = types.AuthConfig`. If we want to have a type alias we should keep the name the same.

Reply:
Ok.
I guess I do not like `AuthConfig` as a name, but that's fine, let's keep it the same.

Reply:
Actually, do you mind if we keep `Credentials`?
`AuthConfig` is really a bad name... If anything, it does evoke https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/7/html/system-level_authentication_guide/authconfig-install#authconfig-install instead of what it really is... just credentials...

It also "makes sense" that the `CredentialsStore` would store `Credentials`.